### PR TITLE
Use only tuple to promote error handling

### PR DESCRIPTION
**Pull request by paramaggarwal**
The Node.js `callback(err, res)` pattern makes sure that the error object is explicitly available and needs to be checked first before expecting the response object. Hence, ignoring the error object requires intentionally ignoring the first argument.

With the goal of making it simpler to adopt better error handling practices, the de-facto approach of this library is now `[err, res]` compared to the `[res, err]` and `{ error, result }` formats in previous versions.

The need for a custom error handler does not exist anymore, now that the error object itself is always available on `[error, result]`

fixes https://github.com/Coly010/notry/issues/3

### CODE REVIEW

**Comment by paramaggarwal** [comment]
Also, small nit: please consider renaming the repo to `no-try` as that is the name of the npm package. 😄

**Comment by Coly010** [comment]
Looks good! 

I'm thinking that we should remove the object destructuring object as you suggested. I've run into issues with it and shadowed variable names.

If you want to tackle that too, that would be awesome!

Minor note, can you add a larger commit message to include the words 

`BREAKING CHANGE`

This will allow semantic release to correctly version the package.

https://github.com/semantic-release/semantic-release#commit-message-format

**Review by Coly010 — APPROVED** [review]
Looks good! Good job!

File: src/index.ts [hunk 1]
```diff
@@ -1,11 +1,11 @@
-export type NoTryResult<T> = { result: T; error: Error };
+export type NoTryTupleResult<T> = [Error, T];
```
Comment (mohsinulhaq) [inline src/index.ts:1]: now that there is no other type, why did we not name it `NoTryResult` itself?

Comment (paramaggarwal) [reply src/index.ts:1]: Good point, feel free to raise a PR!

**Pull request merged**